src/components/InputPanel.tsx

```diff
@@ -1,102 +1,171 @@
+    inputFiles.length
+  );
 
   const inputType = useMemo(() => {
     if (selectedInputFile) {
-      if (selectedInputFile.name.endsWith('.json')) {
+      if (selectedInputFile.name.endsWith(".json")) {
         return INPUT_TYPES.json;
-      } else if (selectedInputFile.name.endsWith('.xml')) {
+      } else if (selectedInputFile.name.endsWith(".xml")) {
         return INPUT_TYPES.xml;
-      } else if (selectedInputFile.name.endsWith('.csv')) {
+      } else if (selectedInputFile.name.endsWith(".csv")) {
         return INPUT_TYPES.csv;
       }
     }
   }, [selectedInputFile]);
 
-  const changeToInputView = useCallback(() => setView(views.inputs), [setView]);
+  const changeView = useCallback((view: ViewType) => setView(view), [setView]);
 
-  const updateSelectedInputFile = useCallback((input: string) => {
-    if (selectedInputFileIndex !== undefined) {
-      setInputFiles(inputFiles.map((inputFile, index) => {
-        if (index === selectedInputFileIndex) {
-          return { ...inputFile, contents: input }
-        }
-        return inputFile;
-      }))
-    }
-  }, [selectedInputFileIndex, setInputFiles, inputFiles]);
+  const updateSelectedInputFile = useCallback(
+    (input: string) => {
+      if (selectedInputFileIndex !== undefined) {
+        setInputFiles(
+          inputFiles.map((inputFile, index) => {
+            if (index === selectedInputFileIndex) {
+              return { ...inputFile, contents: input };
+            }
+            return inputFile;
+          })
+        );
+      }
+    },
+    [selectedInputFileIndex, setInputFiles, inputFiles]
+  );
 
-  const closeSelectedInputFile = useCallback(() => setSelectedInputFileIndex(undefined), []);
+  const closeSelectedInputFile = useCallback(
+    () => setSelectedInputFileIndex(undefined),
+    []
+  );
 
   useEffect(() => {
     if (inputFiles.length !== prevInputFilesLength) {
       setPrevInputFilesLength(inputFiles.length);
       if (inputFiles.length > prevInputFilesLength) {
-        setSelectedInputFileIndex(inputFiles.length-1);
+        setSelectedInputFileIndex(inputFiles.length - 1);
       }
     }
   }, [inputFiles, prevInputFilesLength]);
 
   return (
     <div className={styles.inputPanel}>
       <div className={styles.panelHeader}>
-        <button onClick={changeToInputView} className={`${styles.panelHeaderButton} ${view === views.inputs ? styles.panelHeaderButtonSelected : ''}`}>Input Files</button>
+        {/* <button onClick={changeToInputView} className={`${styles.panelHeaderButton} ${view === views.inputs ? styles.panelHeaderButtonSelected : ''}`}>Input Files</button> */}
+        {Object.keys(views).map((viewType) => (
+          <button
+            key={viewType}
+            className={`${styles.panelHeaderButton} ${
+              view === viewType ? styles.panelHeaderButtonSelected : ""
+            }`}
+            onClick={changeView.bind(null, viewType as ViewType)}
```

Comment:
I like what you've done here to simplify the code as we add more buttons. However, I generally try to avoid using `.bind()` because it returns a new function on every render which can cause extra re-rendering down the line. Maybe it's time to create a `ViewTypeButton` component to which you pass a `viewType` and an `onSelected` prop which expects the `viewType` as a first argument.